new-log-viewer/src/services/LogExportManager.ts

```diff
@@ -0,0 +1,42 @@
+import downloadDecompressedLogs from "../utils/downloadDecompressedLogs";
+
+
+class LogExportManager {
+    private blob: Blob;
+
+    private numChunks: number;
+
+    constructor (numChunks?: number) {
+        this.blob = new Blob();
+        this.numChunks = numChunks ?? 0;
+    }
+
+    getBlob (): Blob {
+        return this.blob;
+    }
+
+    appendChunkData (chunkData: string) {
+        this.blob = new Blob([this.blob,
```

Comment:
Why are we re-creating the Blob on every new chunk append? Can we maintain an array instead, push the data into the array on every new chunk append and generate the Blob only when enough chunks have been received? That way we don't even have to maintain a `numChunksReceived` variable; instead, we can just check against the array's `.length`.